js_modules/dagster-ui/packages/ui-core/src/search/SearchDialog.tsx

```diff
@@ -97,36 +97,31 @@ export const useSearchDialog = () => {
   const openSearch = React.useCallback(() => {
     trackEvent('open-global-search');
     trackEvent('searchOpen');
-    initialize();
     dispatch({type: 'show-dialog'});
-  }, [initialize, trackEvent]);
+  }, [trackEvent]);
 
   React.useEffect(() => {
     __updateSearchVisibility(shown);
   }, [shown]);
 
-  const searchAndHandlePrimary = React.useCallback(
-    async (queryString: string) => {
-      const {queryString: queryStringForResults, results} = await searchPrimary(queryString);
-      dispatch({type: 'complete-primary', queryString: queryStringForResults, results});
-    },
-    [searchPrimary],
-  );
-
-  const searchAndHandleSecondary = React.useCallback(
-    async (queryString: string) => {
-      const {queryString: queryStringForResults, results} = await searchSecondary(queryString);
-      dispatch({type: 'complete-secondary', queryString: queryStringForResults, results});
-    },
-    [searchSecondary],
-  );
-
   const debouncedSearch = React.useMemo(() => {
     return debounce(async (queryString: string) => {
-      searchAndHandlePrimary(queryString);
-      searchAndHandleSecondary(queryString);
+      const [secondaryResults, primaryResults] = await Promise.all([
+        searchSecondary(queryString),
+        searchPrimary(queryString),
+      ]);
```

Comment:
I decided to show them at the same time because I was seeing that one would come in like 10 milliseconds after the other and it created an effect that looked buggy

Reply:
Right now they're still separate webworkers which has some perf benefit since we parallelize searching... but maybe overkill?